pkg/provider/handlers/functions.go

```diff
@@ -231,3 +233,19 @@ func findNamespace(target string, items []string) bool {
 	}
 	return false
 }
+
+func readMemoryLimitFromSpec(spec *specs.Spec) int64 {
+	if spec.Linux != nil {
```

Comment:
Is this correct? Wouldn't it always return 0?

Reply:
Good catch, it wasn't correct as the unit tests indicated lately, I also fixed and simplify that.